tests/bsim/bluetooth/mesh/src/test_blob.c

```diff
@@ -1373,11 +1407,209 @@ static void test_cli_fail_on_no_rsp(void)
+		cli_stop_setup();
+
+		blob_cli.state = BT_MESH_BLOB_CLI_STATE_SUSPENDED;
+		blob_cli.inputs = &blob_cli_xfer.inputs;
+		blob_cli.xfer = &blob_cli_xfer.xfer;
+		blob_cli_xfer.xfer.id = 1;
+		blob_cli.io = &blob_io;
+
+		bt_mesh_blob_cli_resume(&blob_cli);
+
+		if (k_sem_take(&blob_cli_end_sem, K_SECONDS(380))) {
+			FAIL("End CB did not trigger as expected for the cli");
+		}
+
+		ASSERT_TRUE(blob_cli.state == BT_MESH_BLOB_CLI_STATE_NONE);
+
+		break;
+	case BT_MESH_BLOB_XFER_PHASE_SUSPENDED:
+		/* Server will become unresponsive after receiving first chunk */
+		cli_stop_setup();
+
+		blob_cli_prov_and_conf(BLOB_CLI_ADDR);
+
+		err = bt_mesh_blob_cli_send(&blob_cli, &blob_cli_xfer.inputs,
+					    &blob_cli_xfer.xfer, &blob_io);
+		if (err) {
+			FAIL("BLOB send failed (err: %d)", err);
+		}
+
+		if (k_sem_take(&blob_cli_suspend_sem, K_SECONDS(750))) {
+			FAIL("Lost targets CB did not trigger for all expected lost targets");
+		}
+
+		break;
+	default:
+		/* There is no use case to stop in Inactive phase */
+		FAIL();
+	}
+
+	PASS();
+}
+
+static void test_srv_stop(void)
+{
+	if (!recover_settings) {
+		settings_test_backend_clear();
+	}
+
+	bt_mesh_test_cfg_set(NULL, 1000);
+	k_sem_init(&blob_srv_end_sem, 0, 1);
+	k_sem_init(&first_block_wr_sem, 0, 1);
+	k_sem_init(&blob_srv_suspend_sem, 0, 1);
+
+	bt_mesh_device_setup(&prov, &srv_comp);
+
+	switch (expected_stop_phase) {
+	case BT_MESH_BLOB_XFER_PHASE_WAITING_FOR_START:
+		blob_srv_prov_and_conf(bt_mesh_test_own_addr_get(BLOB_CLI_ADDR));
+		bt_mesh_blob_srv_recv(&blob_srv, 1, &blob_io, 0, 1);
+
+		ASSERT_EQUAL(BT_MESH_BLOB_XFER_PHASE_WAITING_FOR_START, blob_srv.phase);
+		break;
+	case BT_MESH_BLOB_XFER_PHASE_WAITING_FOR_BLOCK:
+		ASSERT_EQUAL(BT_MESH_BLOB_XFER_PHASE_WAITING_FOR_START, blob_srv.phase);
+		ASSERT_OK(blob_srv.state.xfer.mode != BT_MESH_BLOB_XFER_MODE_NONE);
+		ASSERT_EQUAL(0, blob_srv.state.ttl);
+
+		k_sem_take(&blob_srv_end_sem, K_SECONDS(500));
+
+		ASSERT_EQUAL(BT_MESH_BLOB_XFER_PHASE_WAITING_FOR_BLOCK, blob_srv.phase);
+		break;
+	case BT_MESH_BLOB_XFER_PHASE_WAITING_FOR_CHUNK:
+		ASSERT_EQUAL(BT_MESH_BLOB_XFER_PHASE_SUSPENDED, blob_srv.phase);
+		ASSERT_EQUAL(0, blob_srv.state.ttl);
+		ASSERT_EQUAL(BLOB_CLI_ADDR, blob_srv.state.cli);
+		ASSERT_EQUAL(1, blob_srv.state.timeout_base);
+		ASSERT_TRUE(BT_MESH_TX_SDU_MAX, blob_srv.state.mtu_size);
+		ASSERT_EQUAL(CONFIG_BT_MESH_BLOB_BLOCK_SIZE_MAX * 2, blob_srv.state.xfer.size);
+		ASSERT_EQUAL(12, blob_srv.state.xfer.block_size_log);
+		ASSERT_EQUAL(1, blob_srv.state.xfer.id);
+		ASSERT_TRUE(blob_srv.state.xfer.mode != BT_MESH_BLOB_XFER_MODE_NONE);
+		/* First block should be already received, second one pending */
+		ASSERT_FALSE(atomic_test_bit(blob_srv.state.blocks, 0));
+		ASSERT_TRUE(atomic_test_bit(blob_srv.state.blocks, 1));
+
+		k_sem_take(&blob_srv_end_sem, K_SECONDS(500));
+
+		ASSERT_EQUAL(BT_MESH_BLOB_XFER_PHASE_WAITING_FOR_CHUNK, blob_srv.phase);
+		break;
+	case BT_MESH_BLOB_XFER_PHASE_COMPLETE:
```

Comment:
Seems handlers for `BT_MESH_BLOB_XFER_PHASE_WAITING_FOR_CHUNK` and `BT_MESH_BLOB_XFER_PHASE_COMPLETE` states are identical.

Reply:
It seems to be leftover from before I fixed client restoration - they were different at some point. I think I can make them fallthrough, I don't see reason to abstract this into funtion

Reply:
I abstracted this into functions, because they have small differences at the end. Same for both client and server parts